packages/api/models/rules.js

```diff
@@ -181,12 +202,41 @@ class Rule extends Manager {
    * @returns {Promise} the response from event source delete
    */
   async deleteKinesisEventSource(item) {
+    if (await this.isEventSourceMappingShared(item)) {
+      return {};
```

Comment:
Any reason to return an empty object here? I think it makes more sense to return undefined (e.g. just `return;`)

Reply:
Ok.  Updated. I just didn't want it to return failure.

Reply:
Actually, eslint will give me error for next return statement if "return;" is used here.
"Async request expected no return value"